File: backend/src/Modules/Auth/Dtos/AuthDto.cs
namespace Trippie.Modules.Auth.Dtos;

public sealed class RegisterDto
{
    public required string Username { get; set; }
    public required string Email { get; set; }
    public required string Password { get; set; }
}

public sealed class AuthResponseDto
{
    public required UserDto User { get; set; }
    public required string Token { get; set; }
    public required DateTime ExpiresAt { get; set; }
    public required string RefreshToken { get; set; }
    public required DateTime RefreshTokenExpiresAt { get; set; }
}

public sealed class LoginDto
{
    public required string Username { get; set; }
    public required string Password { get; set; }
}

public sealed class RefreshTokenRequestDto
{
    public required string RefreshToken { get; set; }
}

public sealed class MeDto
{
    public required string Username { get; set; }
    public required string DisplayName { get; set; }
    public required string Email { get; set; }
    public string? ProfilePhotoUrl { get; set; }
    public List<TripMembershipDto>? Trips { get; set; }
}

public sealed class TripMembershipDto
{
    public required int TripId { get; set; }
    public required string Role { get; set; }
}

File: frontend/src/app/core/feature/auth/dtos/auth.dto.ts
import { UserDto } from './user.dto';

export interface MeDto {
  username: string;
  displayName: string;
  email: string;
  profilePhotoUrl: string | null;
  trips: TripMembershipDto[];
}

export interface TripMembershipDto {
  tripId: number;
  role: 'ADMIN' | 'MEMBER';
}

export interface AuthResponseDto {
  user: UserDto;
  token: string;
  expiresAt: string;
  refreshToken: string;
  refreshTokenExpires: string;
}

export interface LoginDto {
  username: string;
  password: string;
}
